src/price_providers/price_feed.py

```diff
@@ -0,0 +1,31 @@
+from typing import Optional, Tuple
+from src.price_providers.coingecko_pricing import CoingeckoPriceProvider
+from src.price_providers.dune_pricing import DunePriceProvider
+from src.price_providers.moralis_pricing import MoralisPriceProvider
+from src.helpers.config import logger
+
+
+class PriceFeed:
+    """Class encapsulating the different price providers."""
+
+    def __init__(self):
+        self.providers = [
+            CoingeckoPriceProvider(),
+            DunePriceProvider(),
+            MoralisPriceProvider(),
+        ]
+
+    def get_price(
+        self, block_number: int, token_address: str
+    ) -> Optional[Tuple[float, str]]:
+        """Function iterates over list of price provider objects and attempts to get a price."""
+        for provider in self.providers:
+            try:
+                price = provider.get_price(block_number, token_address)
+                if price is not None:
+                    return price, provider.__class__.__name__
+            except Exception as e:
+                logger.error(
```

Comment:
Judging by how common we expect this to be, maybe an info message would be more appropriate compared to the error emitted here.

Reply:
This is actually not common - the last provider in our list would just return None if there's no price. We aren't raising an exception if a price isn't found, its supposed to be for any errors thrown by the price classes. However, I realise that all exceptions are handled within each class and made to return None. So I don't believe there's any way an exception can propogate to lines 27,28. I will still keep it for now.